File: src/main/java/cn/lnkdoc/sdk/uia/instance/AbstractInstance.java
package cn.lnkdoc.sdk.uia.instance;

import cn.lnkdoc.sdk.uia.common.client.IUiaClient;
import cn.lnkdoc.sdk.uia.common.request.IUiaRequest;
import cn.lnkdoc.sdk.uia.common.response.UiaResponse;

/**
 * @author langkye
 * @since 1.0.0.RELEASE
 */
@SuppressWarnings("unchecked")
public abstract class AbstractInstance implements ISdkInstance {
    /**
     * client
     */
    protected IUiaClient client;

    /**
     * extra
     *
     * @param request request
     * @return extra'info
     */
    @Override
    public <T, R> UiaResponse<T> execute(R request) {
        return (UiaResponse<T>) client.execute((IUiaRequest) request);
    }
}

File: src/main/kotlin/cn/lnkdoc/sdk/uia/instance/AbstractInstance.kt
package cn.lnkdoc.sdk.uia.instance

import cn.lnkdoc.sdk.uia.common.client.IUiaClient
import cn.lnkdoc.sdk.uia.common.request.IUiaRequest
import cn.lnkdoc.sdk.uia.common.response.UiaResponse


/**
 * @author langkye
 * @since 1.0.0.RELEASE
 */
@Suppress("UNCHECKED_CAST")
abstract class AbstractInstance : ISdkInstance {
    /**
     * client
     */
    lateinit var client: IUiaClient

    /**
     * extra
     *
     * @param request request
     * @return extra info
     */
    override fun <T, R> execute(request: R): UiaResponse<T> {
        return client.execute<Any>(request as IUiaRequest) as UiaResponse<T>
    }
}

